elodie/geolocation.py

```diff
@@ -146,8 +146,6 @@ def place_name(lat, lon):
                 #  set the most specific as the default.
                 if('default' not in lookup_place_name):
                     lookup_place_name['default'] = address[loc]
-
-    if(lookup_place_name is not {}):
```

Comment:
One of the reasons we check if `lookup_place_name` is not empty is to avoid caching such a value. This could happen for various reasons including bad network connectivity. Caching the value results in always falling back to the default even in cases when the initial problem was transient.

Did `if(lookup_place_name):` not fix your issue?